### DISCUSSION TITLE
Get the state data at a given x/y/z coordinate inside prob.H

### DISCUSSION THREAD

**Discussion by RSuryaNarayan** [AMReX-Combustion/PeleC#810] [Q&A]
Hello 

I am trying to modify the solution using ```void::PeleC::post_time_step``` function in ```prob.H``` (similar to #333 ) and need a function that can tell me what the ```state``` is at a given ```x``` location i.e. roughly the function call must look something like:
```
amrex::Real x_target = 0.5;
auto state_at_xtarget = get_state(x_target);
```
I am trying to see where would a good location be to code this function up. Any pointers are appreciated...

**Comment by baperry2** [comment]
Can you elaborate a bit more on what you're trying to do? From your code snippet, it looks like you're trying to get a slice plane of yz data at a particular value of x? There's machinery to do that in the runtime Diagnostics code (lives in `PelePhysics/Source/Utility`).

Or are you tryin to get point data at a particular (x,y,z) coordinate?

**Reply by RSuryaNarayan** [reply]
I am trying to copy one portion of the domain elsewhere like this (soon after I restart from a checkpoint or a plotfile):
![image](https://github.com/AMReX-Combustion/PeleC/assets/52670594/766473ae-d498-495f-8e51-c3ec2252bd80)

I am using the machinery to modify the state after a time-step using ```PeleC::post_time_step```. Naturally this would need me to get the state at a particular location i.e. ```(x,y,z)``` coordinate and copy that to the window of interest. Hope that makes sense and thanks for your reply!

**Reply by baperry2** [reply]
This will likely be tricky as the two locations will live have data on different processors. My guess is you'll have a better time with this if you are able to do it in terms of indices rather than coordinate values. You may be able to use the AMReX multiblock framework to copy data between the two locations (basically it lets you copy data from an arbitrary box between two multifabs with a specified indexing offset and other transformations). You can see some example code here, but there's not much in terms of documentation, so you'd have to do some digging: https://github.com/AMReX-Codes/amrex/tree/development/Tests/MultiBlock 

Maybe someone will chime in with a simpler approach to doing this with other AMReX functions, but I don't know of one of the top of my head
